ariadne_codegen/plugins/shorter_results.py

```diff
@@ -0,0 +1,179 @@
+
+            # Add all additional imports discovered when generating the client
+            # method.
+            for additional_import in self.extended_imports[stmt.module]:
+                stmt.names.append(ast.alias(name=additional_import))
+
+        return super().generate_client_module(module)
+
+    def generate_client_method(
+        self, method_def: Union[ast.FunctionDef, ast.AsyncFunctionDef]
+    ) -> Union[ast.FunctionDef, ast.AsyncFunctionDef]:
+        """
+        Change the method generated in the client to call the inner field and
+        change the return type.
+        """
+        if len(method_def.body) < 1:
+            return super().generate_client_method(method_def)
+
+        # TODO: Can we assume that the return statement is always the last
+        # statement?
+        return_stmt = method_def.body[-1]
+        if not isinstance(return_stmt, ast.Return):
+            return super().generate_client_method(method_def)
+
+        # Get the originally returned class so we can look at the class
+        # definition to see how many fields it has.
+        if not isinstance(method_def.returns, ast.Name):
+            return super().generate_client_method(method_def)
+
+        return_class = method_def.returns.id
+        if return_class not in self.class_dict:
+            return super().generate_client_method(method_def)
+
+        return_class_ast = self.class_dict[return_class]
+
+        # If we're not a single method data class move on!
+        if len(return_class_ast.body) != 1:
+            return super().generate_client_method(method_def)
+
+        single_field = return_class_ast.body[0]
+
+        # Get the field class name, which might be an optional and if so
+        # represented as an `ast.Subscript`.
+        # We need to update the proper return type to be optional as well so we
+        # store that. The ast in the annotation isn't an evaluated node so we
+        # can't return that as is.
+        if isinstance(single_field.annotation, ast.Name):
+            is_optional = False
+            single_field_class = single_field.annotation.id
+        elif isinstance(single_field.annotation, ast.Subscript):
+            is_optional = True
+            single_field_class = single_field.annotation.slice.id
+        else:
+            return super().generate_client_method(method_def)
+
+        single_field_name = single_field.target.id
+
+        # We only want to expand for classes we created. Some queries will
+        # return a type that has a single field which is of `typing.Any` or
+        # other things not generated this iteration so ignore those.
+        # TODO: Can we ensure this in a better way?
+        try:
+            single_field_class = ast.literal_eval(single_field_class)
+        except ValueError:
+            # Not a literal
+            pass
+
+        if single_field_class not in self.class_dict:
+            return super().generate_client_method(method_def)
```

Comment:
I'm not sure about this but I need to trim the class I get from the ast since it's quoted. However, we can also get types like `typing.Any` which will make `literal_eval` fail.

Further this branch will not enable the plugin for non generated types (as per the docs) but maybe there will be use cases where we want to do this even for `typing.Any`? I think custom scalars work like this? If so would the client method just return `typing.Any`?

Reply:
Makes sense, removed this and added `str` in test!